### PR TITLE
chore(test): Add unit test for KindFilenameMapperUtil

### LINKED ISSUE #1440: JKube Kit : Add unit test for KindFilenameMapperUtil
### Description

[KindFilenameMapperUtil](https://github.com/eclipse/jkube/blob/master/jkube-kit/common/src/main/java/org/eclipse/jkube/kit/common/util/KindFilenameMapperUtil.java) doesn't seem to be covered by any unit tests.

- [ ] Write unit test to verify all required kind to filename mappings are loaded

### PR DESCRIPTION
**Pull request by MMajd**
## Description

Fix #1440 

A unit test for KindFilenameMapperUtil 

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [x] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] I have read the [contributing guidelines](https://www.eclipse.org/jkube/contributing)
 - [x] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [x] My code follows the style guidelines of this project
 - [x] I have performed a self-review of my code
 - [ ] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [ ] I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [ ] I have added tests that prove my fix is effective or that my feature works
 - [ ] New and existing unit tests pass locally with my changes
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

**Comment by anurag-rajawat** [comment]
Please add the license headers to new file by 
```text
mvn -N license:format
```

**Comment by sunix** [comment]
I am not sure that the file `jkube-kit/common/.attach_pid40241` is required.

File: jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/KindFilenameMapperUtilTest.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+import java.util.List;
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class KindFilenameMapperUtilTest {
+    private static final String MAPPING_PROPERTIES = "Pod=pd, pod";
+
+    private static final String VALID_TABLE = "cols=2*,options=\"header\"]" + System.lineSeparator()
+            + "|===" + System.lineSeparator()
```
Comment (sunix) [inline jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/KindFilenameMapperUtilTest.java:29]: the `System.lineSeparator()` are making it hard to read. Are they really required?

Comment (MMajd) [reply jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/KindFilenameMapperUtilTest.java]: I'm really sorry for the delayed response, I have removed the line separators and used new line character, and removed the attached file

File: jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/KindFilenameMapperUtilTest.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            + "a|`cm`, `configmap`\n\n"
+            + "|CronJob\n"
+            + "a|`cj`, `cronjob`\n"
+            + "|===";
+
+    @Test
+    public void shouldLoadMappings() {
+        // given
+        final AsciiDocParser asciiDocParser = new AsciiDocParser();
+        final Map<String, List<String>> serializedMappings = asciiDocParser.serializeKindFilenameTable(new ByteArrayInputStream(VALID_TABLE.getBytes()));
```
Comment (sunix) [inline jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/KindFilenameMapperUtilTest.java:45]: Finally this is just checking the location, not the content, as it is using the same code to parse the files.
What if there is a bug in asciiDocParser.serializeKindFilenameTable not parsing correctly? we won't be able to reveal the problem with this test right?

**Review by sunix — CHANGES_REQUESTED** [review]

File: jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/KindFilenameMapperUtilTest.java (hunk 2 above)
Comment (MMajd) [reply jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/KindFilenameMapperUtilTest.java:45]: I'm really very thankful for you helping me, 
I will look into it

File: jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/KindFilenameMapperUtilTest.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class KindFilenameMapperUtilTest {
+    @Test
+    void shouldLoadMappings() {
+        // given
+        // source: test/resources/**/kind-file-type-mappings-default.ascii
+        Map<String, List<String>> serializedMappings = new HashMap<String, List<String>>() {{
```
Comment (sunix) [inline jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/KindFilenameMapperUtilTest.java:30]: maybe to be renamed with `expectedSerializedMappings`

File: jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/KindFilenameMapperUtilTest.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+            put("BuildConfig", Arrays.asList("bc", "buildconfig"));
+            put("ClusterRole", Arrays.asList("cr", "crole", "clusterrole"));
+            put("ConfigMap", Arrays.asList("cm", "configmap"));
+            put("CronJob", Arrays.asList("cj", "cronjob"));
+        }};
+
+        // source: test/resources/**/kind-file-type-mappings-default.properties
+        Map<String, List<String>> propertiesMappings =  new HashMap<String, List<String>>() {{
```
Comment (sunix) [inline jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/KindFilenameMapperUtilTest.java:38]: maybe to be renamed with `expectedPropertiesMappings`

**Comment by sunix** [comment]
I still can see `jkube-kit/common/.attach_pidxxx` files. I don't thing these are to be part of this PR.

**Review by sunix — APPROVED** [review]

**Review by manusa — APPROVED** [review]

**Pull request merged**